### PR TITLE
Django22 Support

### PR DESCRIPTION
**Pull request by Ayub-Khan**
-Updated Requirement management.
-Updated travis and tox for django22
-Updated torve classifiers
-Version Bump

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{27,35}-django111, py36-django111, quality, docs
+envlist = py{27,35,36}-django111, py35-django{20,21,22},quality, docs
```
Comment (awais786) [inline tox.ini]: python36 with django20 ?

Comment (Ayub-Khan) [reply tox.ini]: I think our main requirement are python3.5 tests. Not sure if we need to check py3.6 too for later versions.

File: Makefile [hunk 2]
```diff
@@ -17,34 +17,34 @@ quality: ## run quality checks using tox
...
+	pip install -r requirements/pip-tools.txt
 
 _upgrade: piptools
-	# Generate pins for test requirements.
-	pip-compile test_requirements.in --upgrade -o test_requirements.txt
-
-	# Delete django pin from test_requirements.txt so that tox can control
+	pip-compile requirements/base.in --upgrade -o requirements/base.txt
+	pip-compile requirements/test.in --upgrade -o requirements/test.txt
+	pip-compile requirements/docs.in --upgrade -o requirements/docs.txt
```
Comment (jmbowman) [inline Makefile]: We should probably add the `--rebuild` option to all these pip-compile lines also.

File: Makefile [hunk 3]
```diff
@@ -17,34 +17,34 @@ quality: ## run quality checks using tox
...
 requirements: piptools ## install test requirements into current env
-	pip-sync test_requirements.txt
-	# `make upgrade` removes Django from test_requirements.txt,
+	pip-sync requirements/test.txt
+	# `make upgrade` removes Django from test.txt,
 	# so we manually install it here.
-	pip install django~=1.11
-	pip install -r docs/requirements.txt
+	pip install -r requirements/base.txt
+	pip install -r requirements/docs.txt
```
Comment (jmbowman) [inline Makefile]: The `pip-sync` line above uninstalls all the extra dependencies from these files each time `make requirements` is run, then they get re-installed here.  The cleanest solution is probably to use `dev.in` and `dev.txt` files like we have elsewhere to collect all the testing and documentation dependencies, and which doesn't have the Django dependency removed.  Then we can just use `pip-sync` on `requirements/dev.txt`.

File: requirements/base.in [hunk 4]
```diff
@@ -0,0 +1,3 @@
+-c constraints.txt
+
+django
```
Comment (jmbowman) [inline requirements/base.in]: Minor complaint: missing newline at end of the file.

File: tox.ini (hunk 1 above)
Comment (jmbowman) [reply tox.ini]: I think we'll eventually want Python 3.6 + Django 2.2 tests, but honestly it depends on whether we upgrade to Ubuntu 18.04 (comes with Python 3.6) next or skip straight to 20.04 (will have either 3.7 or 3.8) after it comes out in April.

**Comment by Ayub-Khan** [comment]
@jmbowman kindly review again.

File: requirements/dev.in [hunk 5]
```diff
@@ -0,0 +1,7 @@
+-c constraints.txt
+
+-r base.txt
+
+-r test.txt
```
Comment (awais786) [inline requirements/dev.in]: why adding tests and docs in dev.in ?

Comment (jmbowman) [reply requirements/dev.in]: We normally do that to allow running any of the tests or doc building commands directly from the development environment without needing to go through tox.  It also helps keep all the dependency versions in sync if there's one requirements file that contains essentially all of the package's dependencies.

Comment (Ayub-Khan) [reply requirements/dev.in]: fro development we need all dependencies of the project.

**Pull request merged**